src/desdemona/functions/sample_functions.clj

```diff
@@ -1,21 +1,27 @@
 (ns desdemona.functions.sample-functions
-  (:require [clojure
-             [walk :refer [postwalk]]]))
+  (:require
+   [clojure.core.match :refer [match]]))
 
-;;; Defines functions to be used by the peers. These are located
-;;; with fully qualified namespaced keywords, such as
-;;; desdemona.functions.sample-functions/format-line
+(defn add-original-wrapper
+  [segment]
+  {:original segment})
 
-(defn transform-segment-shape
-  "Recursively restructures a segment {:new-key [paths...]}"
-  [paths segment]
-  (try (let [f (fn [[k v]]
-                 (if (vector? v)
-                   [k (get-in segment v)]
-                   [k v]))]
-         (postwalk (fn [x] (if (map? x) (into {} (map f x)) x)) paths))
-       (catch Exception e
-         segment)))
+(defn build-row
+  [segment]
+  {:line (str (name (:origin segment)) ": " (-> segment :original :message))})
 
-(defn prepare-rows [segment]
+(defn prepare-rows
+  [segment]
   {:rows [segment]})
+
+(defn message-origin
+  [message]
+  (match [message]
+    [{:parsed {:metadata {:customerIDString _}}}] :falconhose
```

Comment:
(making it work recursively wouldn't be too tricky; that's just clojure.walk/postwalk)